lib/encoder.py

```diff
@@ -796,15 +801,31 @@ def StoreEncoder(self, encoder, workdir=None):
       parameterfile.write(encoder.parameters.ToString())
     encoder.stored = True
 
-  def ReadEncoderParameters(self, hashname):
-    dirname = os.path.join(self.workdir, hashname)
-    with open(os.path.join(dirname, 'parameters'), 'r') as parameterfile:
-      return OptionValueSet(self.context.codec.option_set, parameterfile.read(),
-                            formatter=self.context.codec.option_formatter)
-
-  def AllEncoderFilenames(self):
-    pattern = os.path.join(self.workdir, '*')
-    return [os.path.basename(this_file) for this_file in glob.glob(pattern)]
+  def ReadEncoderParameters(self, dirname):
+    if dirname[0:0] == '/':
```

Comment:
I don't think this is what you want. This will always return the empty string I think? You should probably do [0].

Do you mean to check if the path is relative or absolute? In that case use os.path.isabs.